bundles/org.openhab.binding.neohub/src/main/java/org/openhab/binding/neohub/internal/NeoHubWebSocket.java

```diff
@@ -0,0 +1,238 @@
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.neohub.internal;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.concurrent.ExecutionException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.eclipse.jetty.websocket.api.Session;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketClose;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketConnect;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketError;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketMessage;
+import org.eclipse.jetty.websocket.api.annotations.WebSocket;
+import org.eclipse.jetty.websocket.client.WebSocketClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * Handles the ASCII based communication via web socket between openHAB and NeoHub
+ *
+ * @author Andrew Fiddian-Green - Initial contribution
+ *
+ */
+@NonNullByDefault
+@WebSocket
+public class NeoHubWebSocket extends NeoHubSocketBase {
+
+    private static final int SLEEP_MILLISECONDS = 100;
+    private static final String REQUEST_OUTER = "{\"message_type\":\"hm_get_command_queue\",\"message\":\"%s\"}";
+    private static final String REQUEST_INNER = "{\"token\":\"%s\",\"COMMANDS\":[{\"COMMAND\":\"%s\",\"COMMANDID\":1}]}";
+
+    private final Logger logger = LoggerFactory.getLogger(NeoHubWebSocket.class);
+    private final Gson gson = new Gson();
+    private final WebSocketClient webSocketClient;
+
+    private @Nullable Session session = null;
+    private String responseOuter = "";
+    private boolean responseWaiting;
+
+    /**
+     * DTO to receive and parse the response JSON.
+     *
+     * @author Andrew Fiddian-Green - Initial contribution
+     */
+    private static class Response {
+        @SuppressWarnings("unused")
+        public @Nullable String command_id;
+        @SuppressWarnings("unused")
+        public @Nullable String device_id;
+        public @Nullable String message_type;
+        public @Nullable String response;
+    }
+
+    public NeoHubWebSocket(NeoHubConfiguration config) throws NeoHubException {
+        super(config);
+
+        // initialise and start ssl context factory, http client, web socket client
+        SslContextFactory.Client sslContextFactory = new SslContextFactory.Client();
+        sslContextFactory.setTrustAll(true);
+        HttpClient httpClient = new HttpClient(sslContextFactory);
```

Comment:
Anything preventing the use of the shared framework HTTP client? If having self-singed certificate trust issues, I believe core was prepared for that in 3.3 - see openhab/openhab-core#2622. #11842 does something similar.

Reply:
I did try that initially, but I could not get it to work. But I suppose I was using OH 3.2 so I will have a look again if I could get it working on OH 3.3

Reply:
The current solution is perfectly fine, so let's resolve the comment. My intention was to point you towards a possible solution and assess if it could be applied for this case also. You did, and apparently it's not easy. Unfortunately I don't have time/motivation currently to investigate this deeper and come up with alternative code.